### PR TITLE
Refactor/fully convert jscs to eslint

### PR DESCRIPTION
**Pull request by TerrenceLJones**

### CODE REVIEW

File: test/.eslintrc [hunk 1]
```diff
@@ -2,5 +2,8 @@
   "extends": ["../.eslintrc", "./.globals.json"],
   "env": {
     "mocha": true
-  }
+  },
+  rules: {
+      "new-cap": 0,
```
Comment (paulfalgout) [inline test/.eslintrc]: spacing is off here

File: .eslintrc [hunk 2]
```diff
@@ -30,18 +42,35 @@
...
+    "one-var": [ 2, "never" ],
+    "operator-linebreak": [ 2, "after" ],
     "quotes": [ 2, "single" ],
     "semi": [ 0, "never" ],
+    "space-before-blocks": [ 2, "always" ],
+    "space-before-function-paren": [ 2, "never" ],
+    "space-in-parens": [ 2, "never" ],
+    "space-infix-ops": 2,
+    "space-unary-ops": [ 2,
+      {
```
Comment (paulfalgout) [inline .eslintrc]: I _think_ the spacing is off here, though less so than the other place

**Comment by paulfalgout** [comment]
Not super worried about the spacing in those rc files personally, but just noticed it.
LGTM 👍

**Comment by TerrenceLJones** [comment]
Thanks @paulfalgout...those spacing issues should be fixed in my latest commit.

**Comment by scott-w** [comment]
While we're on the topic - `.eslintrc` is deprecated - we should really be using `.eslintrc.json` or `.eslintrc.yaml` as per the docs.

**Comment by TerrenceLJones** [comment]
Thanks @scott-w. I'll make that needed change.

**Comment by scott-w** [comment]
Interesting, it seems a bunch of stuff depends on `.eslintrc` - feel free to revert them back, it's not exactly critical right now.

**Comment by TerrenceLJones** [comment]
Ok this PR should be ready to go. 🎱

**Comment by scott-w** [comment]
:+1:

**Pull request merged**